concourse/pipelines/templates/build_pipeline-tpl.yml

```
@@ -1,1842 +1,326 @@
-    tag: latest
-    username: _json_key
-    password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
-{% endfor %}{# gp5, gp6 #}
+{% call(x) macros.for_each_config(build_test_pxf_combinations) %}
+    {% include 'resources/pxf-test-image-tpl.yml' %}
+{% endcall %}
 
-{% for gp_ver in range(6, 8) %}
-- name: gpdb[[gp_ver]]-pxf-dev-rocky8-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: gcr.io/data-gpdb-ud/gpdb-pxf-dev/gpdb[[gp_ver]]-rocky8-test-pxf
-    tag: latest
-    username: _json_key
-    password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
-{% endfor %}{# gp6, gp7 #}
-{% set gp_ver = None %}
+    [[macros.resource_registry_image('rpmrebuild-centos7-image', 'gcr.io/data-gpdb-ud/rpmrebuild-centos7', 'latest')]]
+    [[macros.resource_registry_image('rpmrebuild-rocky8-image', 'gcr.io/data-gpdb-ud/rpmrebuild-rocky8', 'latest')]]
 
-- name: gpdb6-pxf-dev-oel7-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: gcr.io/data-gpdb-ud/gpdb-pxf-dev/gpdb6-oel7-test-pxf
-    tag: latest
-    username: _json_key
-    password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
+    [[macros.resource_registry_image('ccp-7-image', 'gcr.io/data-gpdb-public-images/ccp', 'latest')]]
 
-{% set gp_ver = 6 %}
-- name: gpdb[[gp_ver]]-pxf-dev-ubuntu18-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: gcr.io/data-gpdb-ud/gpdb-pxf-dev/gpdb[[gp_ver]]-ubuntu18.04-test-pxf
-    tag: latest
-    username: _json_key
-    password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
-
-{% set gp_ver = None %}
-
-- name: rpmrebuild-centos7-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: gcr.io/data-gpdb-ud/rpmrebuild-centos7
-    tag: latest
-    username: _json_key
-    password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
-
-- name: rpmrebuild-rocky8-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: gcr.io/data-gpdb-ud/rpmrebuild-rocky8
-    tag: latest
-    username: _json_key
-    password: ((ud/pxf/secrets/pxf-cloudbuild-service-account-key))
-
-- name: ccp-7-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: gcr.io/data-gpdb-public-images/ccp
-    tag: latest
-
-- name: google-cloud-sdk-slim-image
-  type: registry-image
-  icon: docker
-  source:
-    repository: google/cloud-sdk
-    tag: 402.0.0-slim
+    [[macros.resource_registry_image('google-cloud-sdk-slim-image', 'google/cloud-sdk', '402.0.0-slim')]]
 
+## ---------- Slack Notification ----------
 {% if slack_notification %}
-- name: slack-alert
-  type: slack-notification
-  source:
-    url: ((ud/pxf/secrets/ud-pipeline-bot-ud-dev-webhook))
-
-- name: notify-releng
-  type: slack-notification
-  source:
-    url: ((ud/pxf/secrets/ud-pipeline-bot-gp-releng-webhook))
+    [[macros.resource_slack_notification('slack-alert', 'ud-pipeline-bot-dev-webhook')]]
```

on the left it has `ud-pipeline-bot-ud-dev-webhook`, so should this be
```suggestion
    [[macros.resource_slack_notification('slack-alert', 'ud-pipeline-bot-ud-dev-webhook')]]
```